Cesium.CodeGen/Extensions/TypeSystemEx.cs

```diff
@@ -15,27 +16,66 @@ internal static class TypeSystemEx
         var methodName = components[1];
 
-        var method = FindMethod(context.AssemblyContext.ImportAssemblies, typeName, methodName);
+        Type[]? types = parametersInfo?.Parameters.Select(x => x.Type.Resolve(context).GetTypeObject()).OfType<Type>().ToArray();
+
+        if (types?.Length != parametersInfo?.Parameters.Count)
+            return null;
+
+        var method = FindMethod(context.AssemblyContext.ImportAssemblies, typeName, methodName, types);
         return method == null ? null : context.Module.ImportReference(method);
     }
 
-    private static MethodInfo? FindMethod(IEnumerable<Assembly> assemblies, string typeName, string methodName)
+    public static MethodReference[] MethodsLookup(this TranslationUnitContext context, string memberName)
+    {
+        var components = memberName.Split("::", 2);
+        if (components.Length != 2)
+            throw new NotSupportedException($"Invalid CLI member name: {memberName}.");
+
+        var typeName = components[0];
+        var methodName = components[1];
+        var methods = FindMethodGroup(context.AssemblyContext.ImportAssemblies, typeName, methodName);
+        return methods == null ? Array.Empty<MethodReference>() : methods.Select(x => context.Module.ImportReference(x)).ToArray();
+    }
+
+    private static MethodInfo? FindMethod(IEnumerable<Assembly> assemblies, string typeName, string methodName, Type[]? parametersType = null)
     {
         foreach (var assembly in assemblies)
         {
-            var method = FindMethod(assembly, typeName, methodName);
+            var method = FindMethod(assembly, typeName, methodName,parametersType );
             if (method != null)
                 return method;
         }
 
         return null;
     }
 
-    private static MethodInfo? FindMethod(Assembly assembly, string typeName, string methodName)
+    private static MethodInfo? FindMethod(Assembly assembly, string typeName, string methodName, Type[]? parametersType = null)
     {
         var type = assembly.GetType(typeName);
 
-        var method = type?.GetMethod(methodName);
+        var method = parametersType is not null ? type?.GetMethod(methodName, parametersType)
+                                                : type?.GetMethod(methodName);
         return method;
     }
+
+    private static MethodInfo[] FindMethodGroup(IEnumerable<Assembly> assemblies, string typeName, string methodName)
+    {
+        return assemblies.SelectMany(x => FindMethodGroup(x, typeName, methodName) ?? Enumerable.Empty<MethodInfo>()).ToArray();
+    }
+
+    private static MethodInfo[]? FindMethodGroup(Assembly assembly, string typeName, string methodName)
+    {
+        return assembly.GetType(typeName)?.GetMethods().Where(x => x.Name == methodName).ToArray();
+    }
+
+    public static Type? GetTypeObject(this TypeReference typeReference)
+    {
+        return Type.GetType($"{typeReference.FullName},{typeReference.Scope}");
+    }
```

Comment:
Unfortunately, this approach won't work: it tries to resolve types using reflection, and types aren't generally available in the compiler process (user types won't resolve via `Type::GetType`).

Why is it even required?

There's the same problem for most of your changes in this file: you're trying to manipulate with types and methods using runtime reflection, which isn't the right approach in a compiler.

Reply:
The reflection code is already there, I'll need to think about it a bit more.